src/NHibernate/Type/TimeAsTimeSpanType.cs

```diff
@@ -43,10 +43,11 @@ public override object Get(DbDataReader rs, int index, ISessionImplementor sessi
 			try
 			{
 				var value = rs[index];
-				if(value is TimeSpan time) //For those dialects where DbType.Time means TimeSpan.
+				if (value is TimeSpan time) //For those dialects where DbType.Time means TimeSpan.
 					return time;
-                
-				return ((DateTime)value).TimeOfDay;
+
+				var dbValue = Convert.ToDateTime(value);
```

Comment:
Is this value really culture specific? Shouldn't invariant culture be provided here?

Reply:
It is actually a variant of ISO8601, see SQLite type doc, followed by the System.Data.SQLite provider (using the variant without `T` but a space instead), so it is neither culture specific, nor culture sensitive.
Ideally this change should be done in every type doing this convert, so also in `AbstractDateTimeType` and `TimeType`. For me, this is out of the scope of this PR, and we should keep it consistent with other types.
(And we would also have to investigate whether the other converts should be switched to invariant culture or not, like those in `NDataReader`.)

Reply:
Not all types suffer from culture specific conversions. DateTime can suffer very badly due to  month and day switch in different cultures. So I don't think introducing new code involving possible DateTime conversion from string should be consistent with other incorrect places.

Reply:
> Ideally this change should be done in every type doing this convert, so also in AbstractDateTimeType

I see two places with different datetime conversions also doesn't look good.. So at least TODO should be added to cover it in next pr

Reply:
Done.

Reply:
https://github.com/nhibernate/nhibernate-core/issues/1173 might be related to this disscussion

Reply:
The inverted case is highly provider dependent, I believe. In most cases, that is the provider which would do the conversion, not NHibernate.